orttraining/orttraining/training_api/onnxruntime_training_c_api.cc

```diff
@@ -278,6 +278,82 @@ ORT_API_STATUS_IMPL(OrtTrainingApis::SaveCheckpoint, _In_ const ORTCHAR_T* check
   API_IMPL_END
 }
 
+ORT_API_STATUS_IMPL(OrtTrainingApis::SetCheckpointProperty, _Inout_ OrtCheckpointState* checkpoint_state,
```

Comment:
This code within this func is actually 'Append' new checkpoint properties, maybe we should use Append or Add instead of Set?